src/hydro_system.hpp

```diff
@@ -583,6 +571,92 @@ void HydroSystem<problem_t>::FlattenShocks(amrex::MultiFab const &q_mf, amrex::M
 	});
 }
 
+// to ensure that physical quantities are within reasonable
+// floors and ceilings which can be set in the param file
+template <typename problem_t>
+void HydroSystem<problem_t>::EnforceLimits(amrex::Real const densityFloor, amrex::Real const pressureFloor, amrex::Real const speedCeiling,
+					   amrex::Real const tempCeiling, amrex::Real const tempFloor, amrex::MultiFab &state_mf)
+{
+
+	amrex::Real const rho_floor = densityFloor; // workaround nvcc bug
+	amrex::Real const P_floor = pressureFloor;
+	auto state = state_mf.arrays();
+
+	amrex::ParallelFor(state_mf, [=] AMREX_GPU_DEVICE(int bx, int i, int j, int k) noexcept {
+		int dindex = HydroSystem<problem_t>::density_index;
+		int p1index = HydroSystem<problem_t>::x1Momentum_index;
+		int p2index = HydroSystem<problem_t>::x2Momentum_index;
+		int p3index = HydroSystem<problem_t>::x3Momentum_index;
+		int eindex = HydroSystem<problem_t>::energy_index;
+		int eint_index = HydroSystem<problem_t>::internalEnergy_index;
+
+		amrex::Real const rho = state[bx](i, j, k, dindex);
+		amrex::Real const vx1 = state[bx](i, j, k, p1index) / rho;
+		amrex::Real const vx2 = state[bx](i, j, k, p2index) / rho;
+		amrex::Real const vx3 = state[bx](i, j, k, p3index) / rho;
+		amrex::Real const vsq = (vx1 * vx1 + vx2 * vx2 + vx3 * vx3);
+		amrex::Real const Etot = state[bx](i, j, k, eindex);
+		amrex::Real const Eint = state[bx](i, j, k, eint_index);
+		amrex::Real const Temp = quokka::EOS<problem_t>::ComputeTgasFromEint(rho, Eint);
```

Comment:
The cooling function should compute the temperature from `Etot - KE`, so I think this should compute the temperature from `Etot` as well, rather than from the auxiliary internal energy.